### PR TITLE
{toolchain} gompi/2015b (REVIEW)

### PR DESCRIPTION
**Pull request by boegel**
depends on ~~#1836~~ (GNU), ~~#1837~~ (Autotools)

**note**: do *not* consider the definition of this common (sub)toolchain final until the PR is merged!

### CODE REVIEW

**Comment by boegel** [comment]
@wpoely86: please review?

File: easybuild/easyconfigs/g/gompi/gompi-2015b.eb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+gccver = '5.1.0'
+binutilsver = '2.25'
+tcver = '%s-binutils-%s' % (gccver, binutilsver)
+
+# compiler toolchain dependencies
+dependencies = [
+    ('GCC', gccver, '-binutils-%s' % binutilsver),
+    ('binutils', binutilsver, '', ('GCC', tcver)),
+    ('OpenMPI', '1.8.5', '', ('GNU', '%s-%s' % (gccver, binutilsver))),
```
Comment (boegel) [inline easybuild/easyconfigs/g/gompi/gompi-2015b.eb]: OpenMPI 1.8.6 should appear very soon, so this should be bumped if it's available before July 1st...

**Comment by wpoely86** [comment]
looks fine

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
Linux RHEL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/20d30ec40a9e6ac80fd2 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
Linux SLES 11_SP2, Intel(R) Xeon(R) CPU E5-2650 v2 @ 2.60GHz, Python 2.6.9
See https://gist.github.com/d0e59c2d0e2a3fcf516c for a full test report.

**Comment by boegel** [comment]
Jenkins: test this please

**Comment by boegel** [comment]
I updated this to use the latest OpenMPI 1.8.6 and hwloc 1.11.0 (just released)

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
Linux RHEL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/9410ff5fb8527321f5a8 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
Linux SLES 11_SP2, Intel(R) Xeon(R) CPU E5-2650 v2 @ 2.60GHz, Python 2.6.9
See https://gist.github.com/d93054a3b20deacc79ac for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
Linux debian 7.8, Intel(R) Xeon(R) CPU            5130  @ 2.00GHz, Python 2.7.3
See https://gist.github.com/caa775922e3ff7393fbb for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
Linux debian 7.8, Intel(R) Xeon(R) CPU            5130  @ 2.00GHz, Python 2.7.3
See https://gist.github.com/896f9a378b69e2f593f6 for a full test report.

File: easybuild/easyconfigs/g/gompi/gompi-2015b.eb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+easyblock = "Toolchain"
+
+name = 'gompi'
+version = '2015b'
+
+homepage = '(none)'
```
Comment (JensTimmerman) [inline easybuild/easyconfigs/g/gompi/gompi-2015b.eb]: should we point to the easybuild readthedocs for toolchains here? None is also fine by me...

File: easybuild/easyconfigs/o/OpenMPI/OpenMPI-1.8.6-GNU-5.1.0-2.25.eb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+# needed for --with-verbs
+osdependencies = [('libibverbs-dev', 'libibverbs-devel')]
+
+sanity_check_paths = {
+    'files': ["bin/%s" % binfile for binfile in ["ompi_info", "opal_wrapper", "orterun"]] +
+             ["lib/lib%s.%s" % (libfile, SHLIB_EXT) for libfile in ["mpi_cxx", "mpi_mpifh",
+                                                                         "mpi", "ompitrace", "open-pal",
+                                                                         "open-rte", "vt", "vt-hyb",
+                                                                         "vt-mpi", "vt-mpi-unify"]] +
```
Comment (JensTimmerman) [inline easybuild/easyconfigs/o/OpenMPI/OpenMPI-1.8.6-GNU-5.1.0-2.25.eb]: move all of these to the left please, letting them hang all at the right there might make them invisible on smaller displays.
```
["lib/lib%s.%s" % (libfile, SHLIB_EXT) for libfile in [
    "mpi_cxx", "mpi_mpifh", "mpi", "ompitrace", "open-pal",
    "open-rte", "vt", "vt-hyb", "vt-mpi", "vt-mpi-unify",
    ]
 ]
```

File: easybuild/easyconfigs/g/gompi/gompi-2015b.eb (hunk 2 above)
Comment (boegel) [inline easybuild/easyconfigs/g/gompi/gompi-2015b.eb]: which page specifically? ;)

Comment (JensTimmerman) [inline easybuild/easyconfigs/g/gompi/gompi-2015b.eb]: http://easybuild.readthedocs.org/en/latest/eb_list_toolchains.html ?

I'm just asking, doesn't have to be changed for me..

Comment (JensTimmerman) [inline easybuild/easyconfigs/g/gompi/gompi-2015b.eb]: Or you could link to this easyconfig here on github ;-)

Comment (boegel) [inline easybuild/easyconfigs/g/gompi/gompi-2015b.eb]: I was half-kidding, we could use a docs page explaining what toolchains are, which ones we have (with versions included), etc.

Comment (fgeorgatos) [inline easybuild/easyconfigs/g/gompi/gompi-2015b.eb]: Hi folks,

if you recall, I offered that page back then when working in the `readthedocs` act,
exactly to make it usable in cases like this. It's the kind of page you need when
you first encounter gompi and you ask yourself: *what the heck is toolchain gompi?*!

File: easybuild/easyconfigs/o/OpenMPI/OpenMPI-1.8.6-GNU-5.1.0-2.25.eb (hunk 3 above)
Comment (fgeorgatos) [inline easybuild/easyconfigs/o/OpenMPI/OpenMPI-1.8.6-GNU-5.1.0-2.25.eb]: +1 for this

it has been an annoyance with readability of all openmpi easyconfigs, all along.

**Comment by fgeorgatos** [comment]
btw. thanks for this PR, it is a very interesting toolchain after @besserox's bugfix included,
esp. for those running big-chunk runs (will propagate the news to my related users asap - ty)

**Comment by boegel** [comment]
@fgeorgatos: don't consider this toolchain final until it has been merged in! (and also see #1695 for `foss/2015b` on top of it)

**Comment by fgeorgatos** [comment]
yes, it's never final until it ends up in master track, anyhow ;-)

**Comment by boegel** [comment]
@gjbex, @backelj, @hajgato: please test?

**Comment by backelj** [comment]
Test report by @backelj
**FAILED**
Build succeeded for 0 out of 4 (4 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.6.6
See https://gist.github.com/56d27badb4099fe71b54 for a full test report.

**Comment by boegel** [comment]
@backelj: you need #1669 and #1693 too, and don't forget to enable `--robot` ;-)

**Comment by backelj** [comment]
@boegel: thanks, I was wondering where the 'potgrond' was ;-)

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
Linux red hat enterprise linux computenode 6.5, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.6.6
See https://gist.github.com/687c8de70b962768bdbb for a full test report.

**Comment by hajgato** [comment]
Test report by @hajgato
**FAILED**
Build succeeded for 2 out of 15 (4 easyconfigs in this PR)
Linux opensuse 13.1, Intel(R) Xeon(R) CPU           W3530  @ 2.80GHz, Python 2.7.6
See https://gist.github.com/adef98a2b83262dfa786 for a full test report.

**Comment by boegel** [comment]
@hajgato:

```
checking for suitable m4... configure: error: No usable m4 in $PATH or /usr/5bin (see config.log for reasons).
```

Can you dive into `config.log` and share the errors in there?

It seems we'll have to include our own M4 as a build dep?

**Comment by hajgato** [comment]
@boegel I need help: runpy ERROR add_dependencies: no module 'zlib/1.2.8-GCC-5.1.0-binutils-2.25' found for dependency

**Comment by boegel** [comment]
@hajgato: you need both #1669 and #1693...

is that the error in `config.log`?

**Comment by hajgato** [comment]
m4 needs as a builddep for GCC-5.1.0-bunutils-2.25

**Comment by backelj** [comment]
Test report by @backelj
**FAILED**
Build succeeded for 10 out of 14 (4 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.6.6
See https://gist.github.com/00e0b19ae185ea0487f8 for a full test report.

**Comment by boegel** [comment]
@backelj: failed checksum for numactl tarball... maybe an interrupted download? Can you try again after renaming the numactl tarball you have?

**Comment by boegel** [comment]
@hajgato: M4 build dep is being included via #1735

**Comment by backelj** [comment]
Test report by @backelj
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.6.6
See https://gist.github.com/1b01fdcf3915ad66d914 for a full test report.

**Comment by hajgato** [comment]
Test report by @hajgato
**FAILED**
Build succeeded for 14 out of 16 (4 easyconfigs in this PR)
Linux opensuse 13.1, Intel(R) Xeon(R) CPU           W3530  @ 2.80GHz, Python 2.7.6
See https://gist.github.com/1ca515a84eec9264cbaa for a full test report.

**Comment by hajgato** [comment]
Test report by @hajgato
**FAILED**
Build succeeded for 0 out of 2 (4 easyconfigs in this PR)
Linux opensuse 13.1, Intel(R) Xeon(R) CPU           W3530  @ 2.80GHz, Python 2.7.6
See https://gist.github.com/61ed78cd448788887952 for a full test report.

**Comment by hajgato** [comment]
Test report by @hajgato
**SUCCESS**
Build succeeded for 16 out of 16 (4 easyconfigs in this PR)
Linux SL 6.4, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.6.6
See https://gist.github.com/ef377a1af639c7ae2e58 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/025b2a59b43e68a07fe6 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/104fcbf4f8a6ca1c00b6 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
Linux red hat enterprise linux computenode 6.5, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.6.6
See https://gist.github.com/bedf1cb4db5f743ad44e for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
Linux centos linux 7.1.1503, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/42b3efdaeae9be43d95e for a full test report.

**Comment by boegel** [comment]
Jenkins: ok to test

**Comment by hajgato** [comment]
Test report by @hajgato
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.6.6
See https://gist.github.com/7bd43179a4633621ba45 for a full test report.

**Comment by boegel** [comment]
Jenkins: test this please

**Comment by boegel** [comment]
Thanks for reviewing/testing everyone!

**Comment by boegel** [comment]
note: the OpenMPI version in `gompi/2015b` was bumped to 1.8.8 via #1841

**Pull request merged**